pkg/apis/pipeline/v1beta1/when_types_test.go

```diff
@@ -307,10 +308,47 @@ func TestApplyReplacements(t *testing.T) {
 			Operator: selection.In,
 			Values:   []string{"barfoo"},
 		},
+	}, {
+		name: "replace array params",
+		original: &WhenExpression{
+			Input:    "$(params.path)",
+			Operator: selection.In,
+			Values:   []string{"$(params.branches[*])"},
+		},
+		replacements: map[string]string{
+			"params.path": "readme.md",
+		},
+		arrayReplacements: map[string][]string{
+			"params.branches": {"dev", "stage"},
+		},
+		expected: &WhenExpression{
+			Input:    "readme.md",
+			Operator: selection.In,
+			Values:   []string{"dev", "stage"},
+		},
+	}, {
+		name: "replace string and array params",
+		original: &WhenExpression{
+			Input:    "$(params.path)",
+			Operator: selection.In,
+			Values:   []string{"$(params.branches[*])", "$(params.files[*])"},
```

Comment:
One additional test-case here for completeness would be `Values` with `string` and `array` vars mixed together:

```go
original: &WhenExpression{
  Input: "$(params.path)",
  Operator: selection.In,
  Values: []string{"$(params.branches[*])", "$(params.matchPath)", "$(params.files[*])"},
},
replacements: map[string]string{
  "params.path": "readme.md",
  "params.matchPath": "foo.txt",
},
arrayReplacements: map[string][]string{
  "params.branches": {"dev", "stage"},
  "params.files": {"readme.md", "test.go"},
},
expected: &WhenExpression{
  Input: "readme.md",
  Operator: selection.In,
  Values: []string{"dev", "stage", "foo.txt", "readme.md", "test.go"},
},
```

Reply:
thanks @sbwsg for an additional unit test, done 👍